cupy/lib/_routines_poly.py

```diff
@@ -104,6 +105,124 @@ def polymul(a1, a2):
     return cupy.convolve(a1, a2)
 
 
+def _astype(x):
```

Comment:
How about naming as `_polyfit_typecast`?

Reply:
If you will not support for inputs of some dtypes, why not raise `NotImplementedError` here?

Reply:
Please address this comment: https://github.com/cupy/cupy/pull/3747#discussion_r471317733